source/lmp/fix_dplr.cpp

```diff
@@ -190,21 +188,46 @@ void FixDPLR::get_valid_pairs(vector<pair<int, int> > &pairs) {
     if (!binary_search(bond_type.begin(), bond_type.end(), bd_type)) {
       continue;
     }
-    if (binary_search(sel_type.begin(), sel_type.end(),
-                      dtype[bondlist[ii][0]]) &&
-        binary_search(dpl_type.begin(), dpl_type.end(),
-                      dtype[bondlist[ii][1]])) {
-      idx0 = bondlist[ii][0];
-      idx1 = bondlist[ii][1];
-    } else if (binary_search(sel_type.begin(), sel_type.end(),
-                             dtype[bondlist[ii][1]]) &&
-               binary_search(dpl_type.begin(), dpl_type.end(),
-                             dtype[bondlist[ii][0]])) {
-      idx0 = bondlist[ii][1];
-      idx1 = bondlist[ii][0];
+    it = find(sel_type.begin(), sel_type.end(), dtype[bondlist[ii][0]]);
```

Comment:
declare `it` here rather than declaring it as a member of the class?

Reply:
OK

Reply:
Done by commit [a7639dd](https://github.com/deepmodeling/deepmd-kit/pull/2580/commits/a7639dd2f13f5dddf3e25f4faab65bd7005313f0).